packages/type-storage/src/fromMetadata.spec.js

```diff
@@ -0,0 +1,25 @@
+import Metadata from '@polkadot/api-codec/Metadata';
+import json from '@polkadot/api-codec/Metadata.rpc.json';
+
+import { fromMetadata } from './fromMetadata';
+
+// Use the pre-generated metadata
+const metadata = new Metadata().fromU8a(Uint8Array.from(json.result));
```

Comment:
You can just use `new Metadata().fromJSON(json.result)`

The `fromJSON` should always allow decoding on the RPCs. Initially it was not the case, but it has been fixed. (i.e. you would normally expect a structure to operate on a, well, json object, in this case it is a number so the codec should handle it.)